### DISCUSSION TITLE
Obsidian Plugin

### DISCUSSION THREAD

**Discussion by JMueller-edu** [LiaScript/LiaScript#217] [Ideas]
Hallo Zusammen,
nutzt sonst noch jemand Obsidian für Notizen?
Wäre ein Plugin hilfreich?

Mit besten Grüßen
Jan Müller

**Comment by andre-dietrich** [comment]
Sounds interesting, I once was also playing around with this idea ... I do not know how much effort this would be, but it should be possible... there are multiple ways I can think of:

1. What is currently working, is the usage of iframes. If you create a note in the LiveEditor:

   https://liascript.github.io/LiveEditor

   then you can also export your content as a data-uri, under Menu >> Export as ... this way the entire content will be encoded within the URL, it seems that gzip might not decode some characters correctly, so the second generated link will work ... You can try this out by clicking onto the link ... one of them will work ;-)

   This ultra long URL can be used within an iframe, in my obsidian it worked:
   
   ``` html
   <iframe  style="width: 100%; aspect-ratio: 10 / 9" src="https://liascript.github.io/course/?data:text/plain;base64,PCEtLQpsaW5rOiAgICAgaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L2NoYXJ0aXN0LmpzL2xhdGVzdC9jaGFydGlzdC5taW4uY3NzCnNjcmlwdDogICBodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvY2hhcnRpc3QuanMvbGF0ZXN0L2NoYXJ0aXN0Lm1pbi5qcwotLT4KCiMgRXhlY3V0YWJsZSBDb2RlCgpBIGRyYXdpbmcgZXhhbXBsZSwgZm9yIGRlbW9uc3RyYXRpbmcgdGhhdCBhbnkgSmF2YVNjcmlwdCBsaWJyYXJ5IGNhbiBiZSB1c2VkLCBhbHNvIGZvciBkcmF3aW5nLgoKYGBgamF2YXNjcmlwdAovLyBJbml0aWFsaXplIGEgTGluZSBjaGFydCBpbiB0aGUgY29udGFpbmVyIHdpdGggdGhlIElEIGNoYXJ0MQpuZXcgQ2hhcnRpc3QuTGluZSgnI2NoYXJ0MScsIHsKICBsYWJlbHM6IFsxLCAyLCAzLCA0XSwKICBzZXJpZXM6IFtbMTAwLCAxMjAsIDE4MCwgMjAwXV0KfSk7CgovLyBJbml0aWFsaXplIGEgTGluZSBjaGFydCBpbiB0aGUgY29udGFpbmVyIHdpdGggdGhlIElEIGNoYXJ0MgpuZXcgQ2hhcnRpc3QuQmFyKCcjY2hhcnQyJywgewogIGxhYmVsczogWzEsIDIsIDMsIDRdLAogIHNlcmllczogW1s1LCAyLCA4LCAzXV0KfSk7CmBgYAo8c2NyaXB0PkBpbnB1dDwvc2NyaXB0PgoKPGRpdiBjbGFzcz0iY3QtY2hhcnQgY3QtZ29sZGVuLXNlY3Rpb24iIGlkPSJjaGFydDEiPjwvZGl2Pgo8ZGl2IGNsYXNzPSJjdC1jaGFydCBjdC1nb2xkZW4tc2VjdGlvbiIgaWQ9ImNoYXJ0MiI+PC9kaXY+Cgo="></iframe>
   ```

   You can style this differently, but it does work ...

2. I do not know how to create obsidian plugins, but an easy way would be to mark content like

   ```` markdown
   ``` liascript
   # The LiaScript content goes here ...
   
   ...
   ```
   ````
   
   and then to transform the content into a base64 encoded string and automatically generate the data-uri for the iframe ...

3. If offline support is required, the LiaScript-Editor can be used for this...

   https://www.npmjs.com/package/@liascript/editor

   We use this package in all of our editors and plugins, it will contain a "website", that can be also loaded within an iframe, but to which you can also send the code to be displayed. This needs to be combined with something link in step 2 ...
